php/hireinsocial/src/HireInSocial/Infrastructure/Doctrine/DBAL/Application/Specialization/DBALSpecializationQuery.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace HireInSocial\Infrastructure\Doctrine\DBAL\Application\Specialization;
+
+use Doctrine\DBAL\Connection;
+use HireInSocial\Application\Query\Specialization\Model\Specialization\FacebookChannel;
+use HireInSocial\Application\Query\Specialization\Model\Specialization;
+use HireInSocial\Application\Query\Specialization\Model\Specializations;
+use HireInSocial\Application\Query\Specialization\SpecializationQuery;
+
+final class DBALSpecializationQuery implements SpecializationQuery
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function all(): Specializations
+    {
+        return new Specializations(
+            ...\array_map(
+                [$this, 'hydrateSpecialization'],
+                $this->connection->fetchAll(
+                    <<<SQL
+                  SELECT 
+                     s.id,
+                     s.slug, 
+                     s.facebook_channel_page_fb_id as 
+                     fb_page_id, s.facebook_channel_group_fb_id as fb_group_id 
+                  FROM his_specialization s 
+                  ORDER BY s.slug
+SQL
+                )
+            )
+        );
+    }
+
+    public function findBySlug(string $slug): ?Specialization
+    {
+        $specialization = $this->connection->fetchAssoc(
+            <<<SQL
+            SELECT 
+               s.id,
+               s.slug, 
+               s.facebook_channel_page_fb_id as fb_page_id, 
+               s.facebook_channel_group_fb_id as fb_group_id
+            FROM his_specialization s
+            WHERE s.slug = :slug
+SQL
+            ,
+            ['slug' => $slug]
+        );
+
+        if (!$specialization) {
+            return null;
+        }
+
+        return $this->hydrateSpecialization($specialization);
+    }
+
+    public function hydrateSpecialization(array $data): Specialization
+    {
+        // TODO: Optimize this, maybe try to merge this into main query or migrate to projections
```

Comment:
For now I dont think this will impact system performance at all but at some point it would be good to optimize this part. It can be optimized by:

a) creating readonly model when specialization is added into the system and updated by new offers. 
b) merge this query into the main one.